addons/product/models/product_template.py

```diff
@@ -824,22 +831,29 @@ def _get_variant_for_combination(self, combination):
 
     @api.multi
     def _get_first_possible_combination(self, parent_combination=None, necessary_values=None):
-        """
-        Iterate the attributes and values in order and stop at the first
-        combination of values that is possible.
+        """See `_get_possible_combinations` (one iteration).
 
-        When encountering an impossible combination, try to change the
-        value of latest attributes first.
-
-        Ignore attributes that have no values.
-
-        Note this method return the same result (empty recordset) if no
+        This method return the same result (empty recordset) if no
         combination is possible at all which would be considered a negative
         result, or if there are no attribute lines on the template in which
         case the "empty combination" is actually a possible combination.
         Therefore the result of this method when empty should be tested
         with `_is_combination_possible` if it's important to know if the
         resulting empty combination is actually possible or not.
+        """
+        return next(self._get_possible_combinations(parent_combination, necessary_values), self.env['product.template.attribute.value'])
+
+    @api.multi
+    def _get_possible_combinations(self, parent_combination=None, necessary_values=None):
```

Comment:
necessary_values=[] and remove fallback later

Reply:
I'm fine either way but we've had issues before when swapping recordset with []. It's obviously fine when only iterating which is the case here though, but that's why I preferred to only fallback in the iteration specifically.